bundle/deploy/state.go

```diff
@@ -0,0 +1,168 @@
+package deploy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/fileset"
+)
+
+const DeploymentStateFileName = "deployment.json"
+const DeploymentStateVersion = 1
+
+type File struct {
+	Path       string `json:"path"`
```

Comment:
Note that this is the **local** path, i.e. including the extension, even for notebooks.